### PR TITLE
PXP-7919 Feat/workspace status

### PR DESCRIPTION
**Pull request by mfshao**
Jira Ticket: [PXP-7919](https://ctds-planx.atlassian.net/browse/PXP-7919)

Add steps to workspace launching page to visualize progress of spinning up a workspace pod

Screen Shot 2021-05-05 at 5 43 56 PM

Require a newer Hatchery, PR: https://github.com/uc-cdis/hatchery/pull/11

Also backward compatiable with older Hatchery (no step progress display)

Deployed and tested in: https://mingfei.planx-pla.net/workspace

### New Features
- Add steps to workspace launching page to visualize progress of spinning up a workspace pod

### CODE REVIEW

File: src/Workspace/index.jsx [hunk 1]
```diff
@@ -112,28 +113,102 @@ class Workspace extends React.Component {
...
     }
     return jupyterIcon;
   }
 
+  getWorkspaceLaunchSteps = (workspaceStatusData) => {
+    if (!(workspaceStatusData.status !== 'Launching' || workspaceStatusData.status !== 'Stopped')
+    || !workspaceStatusData.conditions || workspaceStatusData.conditions.length === 0) {
+      // if status is not 'Launching', or 'Stopped',
+      // or we don't have conditions array, don't display steps bar
+      console.log(workspaceStatusData);
```
Comment (ZakirG) [inline src/Workspace/index.jsx:137]: Did you mean to keep this console log? Also -- I'm surprised the Travis is passing without an eslint rule-disable

File: src/Workspace/index.jsx [hunk 2]
```diff
@@ -173,16 +255,21 @@ class Workspace extends React.Component {
...
       const interval = setInterval(async () => {
-        const status = await this.getWorkspaceStatus();
-        if (this.notebookStates.includes(status)) {
-          this.setState({ notebookStatus: status }, () => {
-            if (this.state.notebookStatus !== 'Launching' &&
-              this.state.notebookStatus !== 'Terminating') {
+        const data = await this.getWorkspaceStatus();
+        if (this.workspaceStates.includes(data.status)) {
+          const workspaceLaunchStepsConfig = this.getWorkspaceLaunchSteps(data);
+          console.log(workspaceLaunchStepsConfig);
```
Comment (ZakirG) [inline src/Workspace/index.jsx:261]: Did you mean to keep this console log?

File: src/Workspace/index.jsx (hunk 1 above)
Comment (mfshao) [reply src/Workspace/index.jsx:137]: oops sorry I didn't push in the lates code 🤦 

I also found that eslint does not always work for some older files, probably an eslint config error

File: src/Workspace/index.jsx [hunk 3]
```diff
@@ -191,14 +278,14 @@ class Workspace extends React.Component {
...
-    this.setState({ notebookIsfullpage: false });
+    this.setState({ workspaceIsfullpage: false });
     // terminate workspace
     this.terminateWorkspace();
   }
 
   handleFullpageButtonClick = () => {
     this.setState({
-      notebookIsfullpage: !this.state.notebookIsfullpage,
+      workspaceIsfullpage: !this.state.workspaceIsfullpage,
```
Comment (ZakirG) [inline src/Workspace/index.jsx:286]: Minor (feel free to ignore): Camel case issue in this variable

**Comment by ZakirG** [comment]
Minor UI feedback (feel free to ignore): I think the text `Launching Workspace, this process may take several minutes` might be better in a gray or black color instead of blue, since it is informative and not clickable.

**Comment by mfshao** [comment]
> Minor UI feedback (feel free to ignore): I think the text `Launching Workspace, this process may take several minutes` might be better in a gray or black color instead of blue, since it is informative and not clickable.

I agree, the blue color comes from the antd Spin component. I think we can either keep our 3 dot spinner, or replace the antd Spin with a different icon, because the current icon has 4 different blue colors so it is kind hard to replace

**Review by ZakirG — APPROVED** [review]

**Comment by mfshao** [comment]
> Minor UI feedback (feel free to ignore): I think the text `Launching Workspace, this process may take several minutes` might be better in a gray or black color instead of blue, since it is informative and not clickable.

I think the old spinner looks better here so I'm putting it back

**Pull request merged**